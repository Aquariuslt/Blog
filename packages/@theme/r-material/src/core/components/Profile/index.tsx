import * as React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Grid from '@material-ui/core/Grid';

interface ProfileProps {
  name: string;
  avatar: string;
  subtitle: string;
  description: string;
}

const useStyles = makeStyles({
  avatar: {
    margin: 10,
    width: 60,
    height: 60
  }
});

const Profile: React.ComponentType<ProfileProps> = (props: ProfileProps): JSX.Element => {
  const classes = useStyles();

  return (
    <Grid container justify="center" alignItems="center">
      <Avatar component="div" alt={props.name} src={props.avatar} className={classes.avatar} />
    </Grid>
  );
};

export default Profile;
