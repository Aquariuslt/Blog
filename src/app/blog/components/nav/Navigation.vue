<template>
  <div>
    <md-toolbar>
      <md-button class="md-icon-button">
        <md-icon>menu</md-icon>
      </md-button>
    </md-toolbar>
  </div>
</template>

<script>
  export default {
    name: 'navigation'
  };
</script>

<style scoped>

</style>
