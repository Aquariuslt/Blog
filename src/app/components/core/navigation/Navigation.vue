<!--suppress ALL -->
<template>
  <section>
    <md-toolbar>
      <md-button class="md-icon-button" @click.native="toggleLeftSideNav">
        <md-icon>menu</md-icon>
      </md-button>
      <h2 class="md-title">
        <router-link class="title" to="/home">{{title}}</router-link>
      </h2>
    </md-toolbar>

    <md-sidenav class="md-left" ref="leftSidenav">
      <md-toolbar>
        <div class="md-toolbar-container">
          <h3 class="md-title">{{title}}</h3>
        </div>
      </md-toolbar>
      <md-list>
        <md-list-item v-for="navMenu in navMenus" v-bind:key="navMenu">
          <router-link :to="navMenu.path">
            <md-icon>{{navMenu.meta.icon}}</md-icon>
            <span>{{navMenu.name}}</span>
          </router-link>
        </md-list-item>
      </md-list>
    </md-sidenav>
  </section>
</template>

<script>
  export default {
    name: 'Navigation',
    methods: {
      toggleLeftSideNav(){
        this.$refs.leftSidenav.toggle();
      }
    },
    computed: {
      title: function () {
        return this.$store.state.core.appName;
      },
      navMenus: function () {
        return this.$store.state.core.navMenus;
      }
    }
  }
</script>

<style lang="scss">
  .title {
    text-decoration: none !important;
    color: #FFFFFF !important;
  }
</style>
