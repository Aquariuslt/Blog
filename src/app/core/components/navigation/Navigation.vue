<!--suppress ALL -->
<template>
  <section>
    <md-toolbar>
      <md-button class="md-icon-button" @click.native="toggleLeftSideNav">
        <md-icon>menu</md-icon>
      </md-button>
      <h2 class="md-title">
        <router-link class="title" to="/home">{{title}}</router-link>
      </h2>
    </md-toolbar>

    <md-sidenav class="md-left" ref="leftSidenav">
      <md-toolbar class="md-large">
        <md-layout md-column md-flex="30">
          <md-avatar class="md-avatar-icon md-large">
            <img :src="avator" alt="Avatar">
          </md-avatar>
        </md-layout>
      </md-toolbar>
      <md-list>
        <md-list-item v-for="navMenu in navMenus" v-bind:key="navMenu">
          <router-link :to="navMenu.path">
            <md-icon>{{navMenu.meta.icon}}</md-icon>
            <span>{{navMenu.name}}</span>
          </router-link>
        </md-list-item>

        <md-list-item v-for="externalLinkCategory in externalLinks" v-bind:key="externalLinkCategory">
          <md-icon>links</md-icon>
          <span>{{externalLinkCategory.label}}</span>
          <md-list-expand>
            <md-list>
              <md-list-item v-for="externalLink in externalLinkCategory.links"
                            v-bind:key="externalLink"
                            class="md-inset">
                <a :href="externalLink.url" class="external-link" target="_blank">
                  {{externalLink.displayName}}
                </a>
                <md-tooltip md-direction="bottom">{{externalLink.description}}</md-tooltip>
              </md-list-item>
            </md-list>
          </md-list-expand>
        </md-list-item>
      </md-list>
    </md-sidenav>
  </section>
</template>

<script>
  export default {
    name: 'Navigation',
    methods: {
      toggleLeftSideNav(){
        this.$refs.leftSidenav.toggle();
      }
    },
    computed: {
      title: function () {
        return this.$store.getters.title;
      },
      avator: function (){
        return this.$store.getters.avator;
      },
      navMenus: function () {
        return this.$store.state.core.navMenus;
      },
      externalLinks: function () {
        return this.$store.getters.externalLinks;
      }
    }
  }
</script>

<style lang="scss">
  .title {
    text-decoration: none !important;
    color: #FFFFFF !important;
  }

  .external-link {
    text-decoration: none !important;
    color: #000000 !important;
  }
</style>
