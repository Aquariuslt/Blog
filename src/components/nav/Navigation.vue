<template>
  <div>
    <md-app>
      <md-app-toolbar class="md-primary" md-mode="flexible">
        <div class="md-toolbar-row">
          <md-button class="md-icon-button" @click="menuVisible = !menuVisible">
            <md-icon>menu</md-icon>
          </md-button>
          <span class="md-title">Aquariuslt Blog</span>
        </div>
      </md-app-toolbar>

      <md-app-drawer :md-active.sync="menuVisible">
      </md-app-drawer>

      <md-app-content>
        <slot></slot>
      </md-app-content>
    </md-app>
  </div>
</template>

<script>
  export default {
    name: 'navigation',
    data: () => ({
      menuVisible: false
    })
  };
</script>

<style scoped>

</style>
