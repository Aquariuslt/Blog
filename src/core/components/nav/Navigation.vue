<template>
  <md-app>
    <md-app-toolbar class="md-primary" md-mode="flexible">
      <div class="md-toolbar-row">
        <md-button class="md-icon-button" @click="menuVisible = !menuVisible">
          <md-icon>menu</md-icon>
        </md-button>
        <span class="md-title">{{title}}</span>
      </div>
    </md-app-toolbar>

    <md-app-drawer :md-active.sync="menuVisible">
      <profile
        :author="author"
        :avator="avator"
        :description="description">
      </profile>
      <nav-menu>

      </nav-menu>
    </md-app-drawer>

    <md-app-content>
      <slot></slot>
    </md-app-content>
  </md-app>
</template>

<script>
  import Profile from '@/core/components/nav/Profile';
  import NavMenu from '@/core/components/nav/NavMenu';

  export default {
    components: {
      NavMenu,
      Profile},
    name: 'navigation',
    props: {
      title: String,
      description: String,
      author: String,
      avator: String
    },
    data: () => ({
      menuVisible: false
    })
  };
</script>

<style scoped>
  .md-app {
    min-height: 100vh;
  }
</style>
