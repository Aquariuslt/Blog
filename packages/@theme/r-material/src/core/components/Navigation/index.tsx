import * as React from 'react';
import classnames from 'classnames';
import { makeStyles, useTheme, Theme, StyleRules } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import RoutingItemList from '../RoutingItemList';
import BundledIcon from '../../components/BundledIcon';

import { NavMenu } from '../../stores/navigation.store';
import { ReactNode } from 'react';
import { PropsWithChildren } from 'react';

interface NavigationProps {
  title: string;
  menus: NavMenu[];
  profile?: ReactNode;
}

const drawerWidth = 240;

const useStyles = makeStyles(
  (theme: Theme): StyleRules => ({
    root: {
      display: 'flex'
    },
    appBar: {
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      })
    },
    appBarShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      })
    },
    menuButton: {
      marginRight: theme.spacing(2)
    },
    hide: {
      display: 'none'
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0
    },
    drawerPaper: {
      width: drawerWidth
    },
    drawerHeader: {
      display: 'flex',
      alignItems: 'center',
      padding: '0 8px',
      ...theme.mixins.toolbar,
      justifyContent: 'flex-end'
    },
    drawerSpace: {
      ...theme.mixins.toolbar,
      marginBottom: '4px'
    },
    drawerButton: {
      padding: '8px'
    },
    content: {
      flexGrow: 1,
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      }),
      marginLeft: -drawerWidth
    },
    contentShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      }),
      marginLeft: 0
    }
  })
);

const Navigation: React.ComponentType<NavigationProps> = (
  props: PropsWithChildren<NavigationProps>
): JSX.Element => {
  const theme: Theme = useTheme();
  const classes = useStyles(theme);

  const [open, setOpen] = React.useState(false);

  function handleDrawerOpen(): void {
    setOpen(true);
  }

  function handleDrawerClose(): void {
    setOpen(false);
  }

  return (
    <div className={classes.root}>
      <CssBaseline/>
      <AppBar
        position="fixed"
        className={classnames(classes.appBar, {
          [classes.appBarShift]: open
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="Open drawer"
            onClick={handleDrawerOpen}
            edge="start"
            href=""
            className={classnames(classes.menuButton, open && classes.hide)}
          >
            <BundledIcon type="menu"/>
          </IconButton>
          <Typography variant="h6" color="inherit" noWrap>
            {props.title}
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={open}
        classes={{
          paper: classes.drawerPaper
        }}
      >
        <div className={classes.drawerHeader}>
          {props.profile}
          <IconButton aria-label="Close drawer" onClick={handleDrawerClose} href="" className={classes.drawerButton}>
            <BundledIcon type="chevronLeft"/>
          </IconButton>
        </div>

        {props.menus.map(
          (menu, index): JSX.Element => (
            <RoutingItemList key={index} {...menu} />
          )
        )}
      </Drawer>
      <main
        className={classnames(classes.content, {
          [classes.contentShift]: open
        })}
      >
        <div className={classes.drawerSpace}/>
        {props.children}
      </main>
    </div>
  );
};

export default Navigation;
