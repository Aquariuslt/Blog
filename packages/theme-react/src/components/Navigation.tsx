import * as React from 'react';
import clsx from 'clsx';
import { Link as RouterLink } from 'react-router-dom';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import useScrollTrigger from '@material-ui/core/useScrollTrigger';
import CssBaseline from '@material-ui/core/CssBaseline';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import Drawer from '@material-ui/core/Drawer';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import Zoom from '@material-ui/core/Zoom';
/* material icons */
import MenuIcon from '@material-ui/icons/Menu';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';

import { Profile } from '@blog/common/interfaces/profile';
import { NavigationItem } from '@blog/common/interfaces/navigation';
import { Icon } from '@theme-react/components/Icon';
import { Fab } from '@material-ui/core';

export interface NavigationProps {
  title: string;
  profile: Profile;
  menus: NavigationItem[];
}

const drawerWidth = 240;
const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex'
    },
    appBar: {
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      })
    },
    appBarShift: {
      width: `calc(100% - ${drawerWidth}px)`,
      marginLeft: drawerWidth,
      transition: theme.transitions.create(['margin', 'width'], {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      })
    },
    menuButton: {
      marginRight: theme.spacing(2)
    },
    hide: {
      display: 'none'
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0
    },
    drawerPaper: {
      width: drawerWidth
    },
    drawerHeader: {
      display: 'flex',
      alignItems: 'center',
      padding: theme.spacing(0, 1),
      ...theme.mixins.toolbar,
      justifyContent: 'flex-end'
    },
    content: {
      flexGrow: 1,
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      }),
      maxWidth: '100%',
      minHeight: '100vh',
      backgroundColor: theme.palette.background.paper,
      marginLeft: -drawerWidth
    },
    contentShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen
      }),
      maxWidth: '100%',
      backgroundColor: theme.palette.background.paper,
      marginLeft: 0
    },
    scrollToTop: {
      position: 'fixed',
      bottom: theme.spacing(2),
      right: theme.spacing(2)
    }
  })
);

const ScrollTop: React.FC = (props) => {
  const { children } = props;
  const classes = useStyles();
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 100
  });

  const handleClick = (event: React.MouseEvent<HTMLDivElement>) => {
    const anchor = ((event.target as HTMLDivElement).ownerDocument || document).querySelector('#back-to-top-anchor');

    if (anchor) {
      console.log(`scrolling to:`, anchor);
      anchor.scrollIntoView({ behavior: 'smooth', block: 'center' });
    }
  };

  return (
    <Zoom in={trigger}>
      <div onClick={handleClick} role="presentation" className={classes.scrollToTop}>
        {children}
      </div>
    </Zoom>
  );
};

export const Navigation: React.FC<NavigationProps> = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen(true);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position="fixed"
        className={clsx(classes.appBar, {
          [classes.appBarShift]: open
        })}
      >
        <Toolbar>
          <IconButton
            color="inherit"
            aria-label="open drawer"
            edge="start"
            onClick={handleDrawerOpen}
            className={clsx(classes.menuButton, open && classes.hide)}
          >
            <MenuIcon />
          </IconButton>
          <Typography variant="h6" noWrap>
            {props.title}
          </Typography>
        </Toolbar>
      </AppBar>
      <Drawer
        className={classes.drawer}
        variant="persistent"
        anchor="left"
        open={open}
        classes={{
          paper: classes.drawerPaper
        }}
      >
        <div className={classes.drawerHeader}>
          <IconButton onClick={handleDrawerClose}>
            <ChevronLeftIcon />
          </IconButton>
        </div>
        <Divider />

        <List>
          {props.menus.map((menu) => (
            <ListItem button key={menu.id} component={RouterLink} to={menu.link}>
              <ListItemIcon>
                <Icon type={menu.icon || 'menu'} />
              </ListItemIcon>
              <ListItemText primary={menu.label} />
            </ListItem>
          ))}
        </List>
      </Drawer>

      <main
        className={clsx(classes.content, {
          [classes.contentShift]: open
        })}
      >
        <Toolbar id="back-to-top-anchor" className={classes.drawerHeader} />
        {props.children}
      </main>
      <ScrollTop {...props}>
        <Fab color="primary" size="small" aria-label="scroll back to top">
          <Icon type="up" />
        </Fab>
      </ScrollTop>
    </div>
  );
};
