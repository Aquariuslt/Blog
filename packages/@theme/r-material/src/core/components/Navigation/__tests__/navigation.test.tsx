import * as React from 'react';

import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

import Navigation from '../';
import Profile from '../../Profile';

Enzyme.configure({ adapter: new Adapter() });

describe('@theme/r-material: core/components: Navigation', (): void => {
  it('# should mount Navigation with props.title correctly', (): void => {
    const mockTitle = 'title';
    const wrapper = Enzyme.shallow(<Navigation title={mockTitle} menus={[]}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('# should show SideBar when state.open is true', (): void => {
    const mockTitle = 'title';

    let wrapper = Enzyme.shallow(<Navigation title={mockTitle} menus={[]}/>);

    // simulate on drawer button will trigger sidebar class change
    wrapper.find('[aria-label="core.navigation.open_drawer"]').simulate('click');

    expect(wrapper).toMatchSnapshot();
  });

  it('# should show SideBar when click twice menu', (): void => {
    const mockTitle = 'title';

    let wrapper = Enzyme.shallow(<Navigation title={mockTitle} menus={[]}/>);

    // simulate on drawer button will trigger sidebar class change
    wrapper.find('[aria-label="core.navigation.open_drawer"]').simulate('click');

    wrapper.find('[aria-label="core.navigation.close_drawer"]').simulate('click');
  });

  it('# should show menus when have non-empty menus props', (): void => {
    const mockTitle = 'title';
    const menus = [
      {
        name: 'PostList',
        link: '/to/somewhere',
        icon: 'category',
        priority: 3,
        child: [
          {
            name: 'mock-sub-name',
            link: '/to/some-else'
          },
          {
            name: 'mock-other-sub-name',
            link: '/to/bar'
          }
        ]
      },
      {
        name: 'Tags',
        link: '/to/somewhere',
        icon: 'bookmark',
        priority: 3,
        child: [
          {
            name: 'mock-sub-name',
            link: '/to/some-else'
          },
          {
            name: 'mock-other-sub-name',
            link: '/to/bar'
          }
        ]
      }
    ];

    let wrapper = Enzyme.shallow(<Navigation title={mockTitle} menus={menus}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('# should show profile when have profile props', (): void => {
    const mockTitle = 'title';
    const menus = [
      {
        name: 'PostList',
        link: '/to/somewhere',
        icon: 'category',
        priority: 3,
        child: [
          {
            name: 'mock-sub-name',
            link: '/to/some-else'
          },
          {
            name: 'mock-other-sub-name',
            link: '/to/bar'
          }
        ]
      },
      {
        name: 'Tags',
        link: '/to/somewhere',
        icon: 'bookmark',
        priority: 3,
        child: [
          {
            name: 'mock-sub-name',
            link: '/to/some-else'
          },
          {
            name: 'mock-other-sub-name',
            link: '/to/bar'
          }
        ]
      }
    ];
    const mockProfileProps = {
      name: '',
      avatar: 'https://img.aquariuslt.com/social/avatar.png',
      subtitle: '',
      description: ''
    };

    let wrapper = Enzyme.shallow(
      <Navigation title={mockTitle} menus={menus} profile={<Profile {...mockProfileProps} />}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
