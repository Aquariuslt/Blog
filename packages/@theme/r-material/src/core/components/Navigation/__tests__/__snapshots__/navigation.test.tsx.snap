// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@theme/r-material: core/components: Navigation # should mount Navigation with props.title correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="menu"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        title
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-6"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-7",
      }
    }
    open={false}
    variant="persistent"
  >
    <div
      className="makeStyles-drawerHeader-8"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="Close drawer"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="chevronLeft"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-9"
  >
    <div
      className="makeStyles-drawerHeader-8"
    />
  </main>
</div>
`;

exports[`@theme/r-material: core/components: Navigation # should show SideBar when state.open is true 1`] = `
<div
  className="makeStyles-root-1"
>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2 makeStyles-appBarShift-3"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Open drawer"
        className="makeStyles-menuButton-4 makeStyles-hide-5"
        color="inherit"
        edge="start"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="menu"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        title
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-6"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-7",
      }
    }
    open={true}
    variant="persistent"
  >
    <div
      className="makeStyles-drawerHeader-8"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="Close drawer"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="chevronLeft"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-9 makeStyles-contentShift-10"
  >
    <div
      className="makeStyles-drawerHeader-8"
    />
  </main>
</div>
`;

exports[`@theme/r-material: core/components: Navigation # should show menus when have non-empty menus props 1`] = `
<div
  className="makeStyles-root-1"
>
  <CssBaseline />
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-appBar-2"
    position="fixed"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="Open drawer"
        className="makeStyles-menuButton-4"
        color="inherit"
        edge="start"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="menu"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        color="inherit"
        noWrap={true}
        variant="h6"
      >
        title
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-6"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-7",
      }
    }
    open={false}
    variant="persistent"
  >
    <div
      className="makeStyles-drawerHeader-8"
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="Close drawer"
        href=""
        onClick={[Function]}
      >
        <BundledIcon
          type="chevronLeft"
        />
      </WithStyles(ForwardRef(IconButton))>
    </div>
    <RoutingItemList
      child={
        Array [
          Object {
            "link": "/to/some-else",
            "name": "mock-sub-name",
          },
          Object {
            "link": "/to/bar",
            "name": "mock-other-sub-name",
          },
        ]
      }
      icon="category"
      key="0"
      link="/to/somewhere"
      name="Categories"
      priority={3}
    />
    <RoutingItemList
      child={
        Array [
          Object {
            "link": "/to/some-else",
            "name": "mock-sub-name",
          },
          Object {
            "link": "/to/bar",
            "name": "mock-other-sub-name",
          },
        ]
      }
      icon="bookmark"
      key="1"
      link="/to/somewhere"
      name="Tags"
      priority={3}
    />
  </WithStyles(ForwardRef(Drawer))>
  <main
    className="makeStyles-content-9"
  >
    <div
      className="makeStyles-drawerHeader-8"
    />
  </main>
</div>
`;
